pwiz/data/vendor_readers/Waters/SpectrumList_Waters.cpp

```diff
@@ -230,26 +230,42 @@ PWIZ_API_DECL SpectrumPtr SpectrumList_Waters::spectrum(size_t index, DetailLeve
     }
 
     float minMZ, maxMZ;
-    rawdata_->Info.GetAcquisitionMassRange(ie.function, minMZ, maxMZ);
+    if (isMS && ie.block >= 0 && hasSonarFunctions())
+    {
+        if (config_.combineIonMobilitySpectra)
+        {
+            rawdata_->Info.GetPrecursorMassRange(ie.function, minMZ, maxMZ); // Get the overall quadrupole filter range
+        }
+        else
+        {
+            rawdata_->Info.GetPrecursorMassRange(ie.function, ie.scan, minMZ, maxMZ); // Get the quadrupole filter range this spectrum
+        }
+    }
+    else
+    {
+        rawdata_->Info.GetAcquisitionMassRange(ie.function, minMZ, maxMZ);
+    }
     scan.scanWindows.push_back(ScanWindow(minMZ, maxMZ, xUnit));

```

Comment:
As mentioned in my (tardily sent) email, scan window should definitely be the min and max m/z possible in the spectrum itself (that's the TOF analyzer for SONAR, not quadrupole). It should have nothing to do with isolation. For MS1 SONAR in the uncombined representation, we can't use the isolation window element (same problem we have with plain MSe), so I think we should just put a couple of userParams in the root result spectrum for that representation. We won't (can't) add them for the combined representation because that's what the extra 2 arrays are for.

Reply:
CV update was pretty smooth, except for a term that had been renamed and required a little manual intervention.

Can I leave the userParam stuff to you? I think you have a much clearer vision of what you want there.